lib/moduleEncoding.js

```diff
@@ -305,6 +604,408 @@ async function compressModule(data, options) {
+
+		buffer = Buffer.alloc(data.length + (suffix.suffixSize - originalSuffixSize));
+		data.copy(buffer, 0, 0, data.length - originalSuffixSize - 4 /* CRC-32 */);
+		prefix.moduleEndAddy = prefix.moduleStartAddy + buffer.length - 4;
+	}
+
+	updateModulePrefix(buffer, prefix);
+	updateModuleSuffix(buffer, suffix);
+	updateModuleSha256(buffer);
+	updateModuleCrc32(buffer);
+
+	return buffer;
+}
+
+/**
+ * Create asset module from raw asset data
+ *
+ * @param {Buffer} data Asset data.
+ * @param {Object} [options] Options.
+ * @param {Object} [options.zlib] Compression options (https://nodejs.org/api/zlib.html#zlib_class_options).
+ * @param {Boolean} [options.compress] Compress resulting module (default = true)
+ * @param {String} name Asset name
+ * @returns {Promise<Buffer>}
+ */
+async function createAssetModule(data, name, options) {
+	let hash = crypto.createHash('sha256');
+	hash.update(data);
+	hash = hash.digest();
+
+	const suffixInfo = {
+		extensions: [
+			{
+				type: ModuleInfo.ModuleInfoExtension.HASH,
+				hash: hash,
+				hashType: ModuleInfo.ModuleInfoHashExtensionType.SHA256
+			},
+			{
+				type: ModuleInfo.ModuleInfoExtension.NAME,
+				name: name
+			},
+			{
+				// For compatibility
+				type: ModuleInfo.ModuleInfoExtension.PRODUCT_DATA,
+				productId: 0xffff,
+				productVersion: 0xffff
+			}
+		]
+	};
+	suffixInfo.suffixSize = calculateModuleSuffixSize(suffixInfo);
+
+	let buffer = Buffer.alloc(MODULE_PREFIX_SIZE + suffixInfo.suffixSize + data.length + 4 /* CRC-32 */);
+	// Generate module prefix and suffix
+	const prefixInfo = {
+		moduleStartAddy: 0,
+		moduleEndAddy: buffer.length - 4,
+		moduleFunction: ModuleInfo.FunctionType.ASSET,
+		moduleFlags: ModuleInfo.Flags.DROP_MODULE_INFO
+	};
+
+	data.copy(buffer, MODULE_PREFIX_SIZE);
+	updateModulePrefix(buffer, prefixInfo);
+	updateModuleSuffix(buffer, suffixInfo);
+	updateModuleSha256(buffer);
+	updateModuleCrc32(buffer);
+
+	if (!options || options.compress === undefined || options.compress) {
+		buffer = await compressModule(buffer);
+	}
+
+	return buffer;
+}
+
+
+/**
+ * Unwrap asset module to raw asset data
+ *
+ * @param {Buffer} data Asset module
+ * @returns {Promise<Buffer>}
+ */
+async function unwrapAssetModule(data) {
+	const parser = new HalModuleParser();
+	let { prefixInfo: prefix } = await parser.parsePrefix({ fileBuffer: data });
+	let { suffixInfo: suffix } = await parser.parseSuffix({ fileBuffer: data });
+	if (prefix.moduleFunction !== ModuleInfo.FunctionType.ASSET) {
+		throw new RangeError('Invalid module function type');
+	}
+	if (!(prefix.moduleFlags & ModuleInfo.Flags.DROP_MODULE_INFO)) {
+		throw new RangeError('Asset modules should contain DROP_MODULE_INFO flag');
+	}
+	if (prefix.moduleFlags & ModuleInfo.Flags.COMPRESSED) {
+		data = await decompressModule(data);
+	}
+	({ prefixInfo: prefix } = await parser.parsePrefix({ fileBuffer: data }));
+	({ suffixInfo: suffix } = await parser.parseSuffix({ fileBuffer: data }));
```

Comment:
With the `DROP_MODULE_INFO` flag set, `decompressModule()` essentially just copies the prefix and suffix data from the compressed module so it shouldn't be necessary to parse those again.